### PR TITLE
runtime: add FANUC's KAREL support

### PR DESCRIPTION
**Pull request by kirillmorozov**
I've noticed that thanks to @KnoP-01 Vim provides a nice experience when working with code for industrial robots(ABB, KUKA) but syntax highlighting for FANUC's KAREL language does not come out of the box.

This PR fixes this omission by adding KAREL syntax highlighting originally developed by @unreal and filetype plugin originally developed by @KnoP-01.

### CODE REVIEW

File: runtime/syntax/karel.vim [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+" Credits:     Jay Strybis for the initial implementation and Patrick Knosowski
+"              for a couple of fixes.
+
+if exists("b:current_syntax")
+  finish
+endif
+
+if version < 600
+  syntax clear
+endif
```
Comment (chrisbra) [inline runtime/syntax/karel.vim]: You can get rid of this block.

Comment (kirillmorozov) [reply runtime/syntax/karel.vim]: Done.

**Comment by chrisbra** [comment]
Thanks, do we also need a filetype plugin (e.g. for setting the comment character)?

**Comment by KnoP-01** [comment]
@kirillmorozov have you noticed the additions I made to Fanuc Karel syntax and the ftplugin? Maybe those are worth adding as well?

**Comment by kirillmorozov** [comment]
Hi @KnoP-01, yes, except those that you've made 7 minutes ago. I have even added you to credits section because of that.

**Comment by KnoP-01** [comment]
I just added my local ftplugin. It's pretty basic, so if Chris is fine with it we could add this as well. 

Actually I don't nurse the Fanuc files as much as I do for Kuka and ABB. So I'm happy that you step in to take the maintainer role.

**Comment by chrisbra** [comment]
that would be fine from my side

**Comment by kirillmorozov** [comment]
Ok, I've added the ftplugin as well and changed PR title and description to reflect it.

File: runtime/ftplugin/karel.vim [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+" Maintainer:  Kirill Morozov <kirill@robotix.pro>
+" Credits:     Patrick Meiser-Knosowski for the initial implementation.
+
+if exists("b:did_ftplugin")
+  finish
+endif
+
+let b:did_ftplugin = 1
+let s:keepcpo = &cpo
+set cpo&vim
```
Comment (chrisbra) [inline runtime/ftplugin/karel.vim]: you can drop this block (and the restoration at the end of the file)

Comment (KnoP-01) [reply runtime/ftplugin/karel.vim]: including line 11? I see we don't need keepcpo since there is no line continuation, but line 11...?

Comment (kirillmorozov) [reply runtime/ftplugin/karel.vim]: I kept line 11 and removed everything else.

Comment (chrisbra) [reply runtime/ftplugin/karel.vim]: thanks. Exactly like this :)

Comment (kirillmorozov) [reply runtime/ftplugin/karel.vim]: @chrisbra, is there a help page I could read that describes typical steps required to create a ftplugin as well as mentions the need(or lack thereof) for `set cpo&vim`?

I saw that earlier ftplugins use it while newer ones do not.

Comment (chrisbra) [reply runtime/ftplugin/karel.vim]: We don't have this very specific, only `:h write-filetype-plugin`. When I review a PR, I try to check those settings and if it looks like it isn't needed, I give the advice to leave those things out. But older runtime files may still have it.

**Comment by KnoP-01** [comment]
thanks @kirillmorozov for your initiative. Thanks Chris for inclusion.

**Pull request closed without merging**